### PR TITLE
Constrain rate dynamics to [-500,500]

### PR DESCRIPTION
**Pull request by avsaase**
With `rate_dynamics_end_sensitivity` above 100, rate dynamics can push `rcCommand` outside of the conventional [-500,500] range. This causes setpoints higher than the configured rate and also leads to a sort of bounceback as `rcCommand` overshoots the stick position and is then corrected. I realize that this last point is perhaps by design so this PR is up for discussion.

@Quick-Flash WDYT?

### CODE REVIEW

**Comment by Quick-Flash** [comment]
This is by design, but I have tried thinking of ways to minimize bounceback from things like this. Particularly the potential bounceback as you go back to center stick. Im not sure that such an abrupt approach is the best way. But perhaps itd be alright. Lemme know how it feels.

**Comment by DzikuVx** [comment]
@avsaase  `master` is now building, please merge it your branch

File: src/main/flight/rate_dynamics.c [hunk 1]
```diff
@@ -73,13 +74,14 @@ FAST_CODE int applyRateDynamics(int rcCommand, const int axis, const float dT) {
...
-        rcCommand = rcCommand + iterm[axis]; // add the iterm to the rcCommand
+        rcCommand += iterm[axis]; // add the iterm to the rcCommand
 
         dterm_centerStick = inverseRcCommandPercent * (lastRcCommandData[axis] - rcCommand) * calculateK(currentControlRateProfile->rateDynamics.weightCenter / 100.0f, dT); // valid dterm values are between 0-95
         dterm_endStick = rcCommandPercent * (lastRcCommandData[axis] - rcCommand) * calculateK(currentControlRateProfile->rateDynamics.weightEnd / 100.0f, dT);
         dterm = dterm_centerStick + dterm_endStick;
-        rcCommand = rcCommand + dterm; // add dterm to the rcCommand (this is real dterm)
+        rcCommand += dterm; // add dterm to the rcCommand (this is real dterm)
 
+        rcCommand = constrain(rcCommand, -500, 500);
```
Comment (DzikuVx) [inline src/main/flight/rate_dynamics.c:84]: @avsaase I like this limit. But I'd set it slightly higher than 500. Let's say 550 so it still allows to raise if needed, but not go crazy values

**Comment by avsaase** [comment]
Playing around with the demo on https://stoot.tech/emu-rate-dynamics.html, I notice the output can go outside of the [-500, 500] range with a bounce even when all sensitivities are below 100. For example with center sensitivity at 60 and the rest at defaults. Should we add some anti-windup logic?

**Comment by avsaase** [comment]
I put together a quick Java program to simulate rate dynamics from joystick input so I can get a better feeling for how it works. So far the highest `rcCommand` I have been able to produce is 541. Now that I can visualize* the rate dynamics I will experiment with some anti-windup strategies to see if I can get rid of this bounce.

* it's just a crappy command-line program for now. I will make a GUI later.

**Comment by Quick-Flash** [comment]
> I put together a quick Java program to simulate rate dynamics from joystick input so I can get a better feeling for how it works. So far the highest `rcCommand` I have been able to produce is 541. Now that I can visualize* the rate dynamics I will experiment with some anti-windup strategies to see if I can get rid of this bounce.
> 
> *it's just a crappy command-line program for now. I will make a GUI later.

I have thought of doing some sort of "anti-windup" for the iterm portion of the rate dynamics, but I don't know the best option for doing it. So i never really ended up playing around with it.

**Comment by avsaase** [comment]
I-term relax is the first thing that comes to my mind, but I haven't tried it yet so I don't know if it will work. I'll close this PR now and open a new one if I find some improvements.

**Comment by Quick-Flash** [comment]
> I-term relax is the first thing that comes to my mind, but I haven't tried it yet so I don't know if it will work. I'll close this PR now and open a new one if I find some improvements.

Lemme know what you come up with.

**Pull request closed without merging**